RecoBTag/DeepFlavour/interface/c_pf_features_converter.h

```diff
@@ -0,0 +1,263 @@
+            //c_pf_features.vertexRef_mass=c_pf->vertexRef()->p4().M();
+
+            c_pf_features.puppiw = puppiw;
+
+            const auto & pseudo_track =  *c_pf->bestTrack();
+            auto cov_matrix = pseudo_track.covariance();
+
+            c_pf_features.dptdpt =    deep::catch_infs_and_bound(cov_matrix[0][0],0,-1,1);
+            c_pf_features.detadeta=   deep::catch_infs_and_bound(cov_matrix[1][1],0,-1,0.01);
+            c_pf_features.dphidphi=   deep::catch_infs_and_bound(cov_matrix[2][2],0,-1,0.1);
+
+            /*
+             * what makes the most sense here if a track is used in the fit... cerntainly no btag
+             * for now leave it a t zero
+             * infs and nans are set to poor quality
+             */
+            //zero if pvAssociationQuality ==7 ?
+            c_pf_features.dxydxy =    deep::catch_infs_and_bound(cov_matrix[3][3],7.,-1,7);
+            c_pf_features.dzdz =      deep::catch_infs_and_bound(cov_matrix[4][4],6.5,-1,6.5); 
+            c_pf_features.dxydz =     deep::catch_infs_and_bound(cov_matrix[3][4],6.,-6,6); 
+            c_pf_features.dphidxy =   deep::catch_infs(cov_matrix[2][3],-0.03);
+            c_pf_features.dlambdadz=  deep::catch_infs(cov_matrix[1][4],-0.03);
+
+
+            c_pf_features.BtagPf_trackMomentum   =deep::catch_infs_and_bound(track_info.getTrackMomentum(),0,0 ,1000);
+            c_pf_features.BtagPf_trackEta        =deep::catch_infs_and_bound(track_info.getTrackEta()   ,  0,-5,5);
+            c_pf_features.BtagPf_trackEtaRel     =deep::catch_infs_and_bound(track_info.getTrackEtaRel(),  0,-5,15);
+            c_pf_features.BtagPf_trackPtRel      =deep::catch_infs_and_bound(track_info.getTrackPtRel(),   0,-1,4);
+            c_pf_features.BtagPf_trackPPar       =deep::catch_infs_and_bound(track_info.getTrackPPar(),    0,-1e5,1e5 );
+            c_pf_features.BtagPf_trackDeltaR     =deep::catch_infs_and_bound(track_info.getTrackDeltaR(),  0,-5,5 );
+            c_pf_features.BtagPf_trackPtRatio    =deep::catch_infs_and_bound(track_info.getTrackPtRatio(), 0,-1,10 );
+            c_pf_features.BtagPf_trackPParRatio  =deep::catch_infs_and_bound(track_info.getTrackPParRatio(),0,-10,100);
+            c_pf_features.BtagPf_trackSip3dVal   =deep::catch_infs_and_bound(track_info.getTrackSip3dVal(), 0, -1,1e5 );
+            c_pf_features.BtagPf_trackSip3dSig   =deep::catch_infs_and_bound(track_info.getTrackSip3dSig(), 0, -1,4e4 );
+            c_pf_features.BtagPf_trackSip2dVal   =deep::catch_infs_and_bound(track_info.getTrackSip2dVal(), 0, -1,70 );
+            c_pf_features.BtagPf_trackSip2dSig   =deep::catch_infs_and_bound(track_info.getTrackSip2dSig(), 0, -1,4e4 );
+            c_pf_features.BtagPf_trackDecayLen   =0;
+            c_pf_features.BtagPf_trackJetDistVal =deep::catch_infs_and_bound(track_info.getTrackJetDistVal(),0,-20,1 );
+            c_pf_features.BtagPf_trackJetDistSig =deep::catch_infs_and_bound(track_info.getTrackJetDistSig(),0,-1,1e5 );
```

Comment:
common parts with c_pf_packed_features_converter should be moved to one method to avoid copy-pastes

Reply:
ok